dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -547,17 +546,24 @@ private DhisDataValueStatusResponse importDataValues(Settings settings, String u
         HttpUriRequest request = generatePostRequest(settings, uri, json);
         LOGGER.debug(String.format("Initiating request to create resource: %s, request: %s", json, request.toString()));
         CloseableHttpResponse response = getResponseForRequest(request);
-        LOGGER.debug(String.format("Received response to create resource: %s, request: %s", json, response));
+        DhisDataValueStatusResponse status;
 
         try (InputStream content = getContentForResponse(response)) {
-            return new ObjectMapper().readValue(content, DhisDataValueStatusResponse.class);
+            String contentString = IOUtils.toString(content);
+            LOGGER.debug(String.format("Received response to create resource: %s, request: %s", json, contentString));
+            status = new ObjectMapper().readValue(contentString, DhisDataValueStatusResponse.class);
+            if (status.getStatus() == DhisStatus.ERROR) {
+                String msg = String.format("Error in DHIS2 status response, error details: %s", contentString);
+                throw new DhisWebException(msg);
+            }
```

Comment:
can you create separate method for this, because this is almost the same code as above